src/foremast/utils/awslambda.py

```diff
@@ -99,22 +100,59 @@ def add_lambda_permissions(function='',
     session = boto3.Session(profile_name=env, region_name=region)
     lambda_client = session.client('lambda')
     response_action = None
+    prefixed_sid = 'foremast-' + statement_id
 
     add_permissions_kwargs = {
         'FunctionName': function,
-        'StatementId': statement_id,
+        'StatementId': prefixed_sid,
         'Action': action,
         'Principal': principal,
     }
+
     if source_arn:
         add_permissions_kwargs['SourceArn'] = source_arn
 
     try:
         lambda_client.add_permission(**add_permissions_kwargs)
-        response_action = 'Add permission with Sid: {}'.format(statement_id)
+        response_action = 'Add permission with Sid: {}'.format(prefixed_sid)
     except boto3.exceptions.botocore.exceptions.ClientError as error:
         LOG.debug('Add permission error: %s', error)
         response_action = "Did not add permissions"
 
-    LOG.debug('Related StatementId (SID): %s', statement_id)
+    LOG.debug('Related StatementId (SID): %s', prefixed_sid)
     LOG.info(response_action)
+
+def remove_all_lambda_permissions(app_name='', env='', region='us-east-1'):
+    """removes all foremast-* permissions from lambda
+
+    Args:
+        app_name (str): Application name
+        env (str): AWS environment
+        region (str): AWS region
+    """
+    session = boto3.Session(profile_name=env, region_name=region)
+    lambda_client = session.client('lambda')
+    legacy_prefix = app_name + "_"
+    foremast_prefix = "foremast-"
+
+    lambda_arn = get_lambda_arn(app_name, env, region)
+    lambda_alias_arn = get_lambda_alias_arn(app_name, env, region)
+    arns = (lambda_arn, lambda_alias_arn)
+
+    for arn in arns:
+        try:
+            response = lambda_client.get_policy(FunctionName=arn)
+        except boto3.exceptions.botocore.exceptions.ClientError as error:
+            LOG.info("No policy exists for function %s, skipping deletion", arn)
+            LOG.debug(error)
+            return
```

Comment:
This could return immediately on the first ARN. Do we want that behaviour, or should we `continue` to the next ARN?

If we do want to return immediately, we should probably return `False` to indicate no changes. Then at the end return `True` to indicate changes made.